src/mongoChangeStream.test.ts

```diff
@@ -493,6 +517,43 @@ describe('syncing', () => {
     assert.equal(cursorError, false)
   })
 
+  test('should process records via change stream', async () => {
+    const { coll, db } = await getConns()
+    const sync = await getSync()
+    await initState(sync, db, coll)
+
+    let cursorError = false
+    sync.emitter.on('cursorError', () => {
+      cursorError = true
+    })
+    const processed: any[] = []
+    const processRecords = async (docs: ChangeStreamDocument[]) => {
+      for (const doc of docs) {
+        await setTimeout(5)
+        processed.push(doc)
+        if (processed.length === 200) {
+          sync.pausable.pause()
+          global.setTimeout(sync.pausable.resume, 1000)
+        }
+      }
+    }
+    const changeStream = await sync.processChangeStream(processRecords, {
+      batchSize: 100,
+    })
+    // Start
+    changeStream.start()
+    await setTimeout(ms('1s'))
+    // Update records
+    coll.updateMany({}, { $set: { createdAt: new Date('2022-01-01') } })
```

Comment:
Should this have an `await`?

Reply:
No. Almost all of the `updateMany` calls are not awaited.